src/Icicle/Source/Checker/Prim.hs

```diff
@@ -81,6 +86,9 @@ primLookup' p
   fNum f
    = f1 (\a at -> uncurry (FunctionType [a] [CIsNum at]) (f at))
 
+  fBool f
+   = f1 (\a at -> uncurry (FunctionType [a] [CEquals BoolT at]) (f at))
```

Comment:
this can just call f with BoolT - it doesn't need a constraint for that.
in fact the type can just be
```FunctionType [] [] [BoolT, BoolT] BoolT```